From b306f79f7ba0f3b9eaab711134fe7d09bc10fa78 Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 13:12:52 -0400
Subject: [ethernet] i40e/i40evf: Add new HW link info variable an_enabled and function update_link_info

Message-id: <1409663591-32531-130-git-send-email-sassmann@redhat.com>
Patchwork-id: 90605
O-Subject: [RHEL7.1 PATCH 129/148] i40e/i40evf: Add new HW link info variable an_enabled and function update_link_info
Bugzilla: 1091127
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Catherine Sullivan <catherine.sullivan@intel.com>

Add a new variable, hw.phy.link_info.an_enabled, to track whether autoneg is
enabled.  Also add a new function update_link_info that will update that
variable as well as calling get_link_info to update the rest of the link info.
Also add get_phy_capabilities to support this.

Change-ID: I5157ef03492b6dd8ec5e608ba0cf9b0db9c01710
Signed-off-by: Catherine Sullivan <catherine.sullivan@intel.com>
Tested-by: Jim Young <jamesx.m.young@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit 8109e1232b3e5322415a9b5e09951617c5fae277)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/i40evf/i40e_type.h b/drivers/net/ethernet/intel/i40evf/i40e_type.h
index 23cd18b..5b6e955 100644
--- a/drivers/net/ethernet/intel/i40evf/i40e_type.h
+++ b/drivers/net/ethernet/intel/i40evf/i40e_type.h
@@ -166,6 +166,7 @@ struct i40e_link_status {
  u8 an_info;
  u8 ext_info;
  u8 loopback;
+ bool an_enabled;
  /* is Link Status Event notification to SW enabled */
  bool lse_enable;
  u16 max_frame_size;
-- 
1.7.1